docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,825 @@
+| Subnets             | List of CIDRs used for the pod network across the cluster.Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed. The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `ipam.mode` is `Disabled`.                                                                                                                                                                                                                                                                                                                        | Yes      |
+| ExcludeSubnets      | List of CIDRs removed from the specified CIDRs in `subnets`.The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.                                                                                                                                                                                                                                                                                                                                                                                                    | Yes      |
+| IPAM                | Contains IPAM-related configuration, similar to Layer2  & Layer3 topologies.<br>Consist of the following fileds:<br>`mode`:<br> When `Enabled`, OVN-Kubernetes will apply IP configuration to the SDN infra and assign IPs from the selected subnet to the pods.<br>When `Disabled`, OVN-Kubernetes only assign MAC addresses and provides layer2 communication, enable users configure IP addresses to the pods.<br>`lifecycle`:<br> When `Persistent` enable workloads have persistent IP addresses. For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots). |          |          |
+
+#### Suggested API validations
+- `Role`:
+    - Required.
+    - When `topology=Localnet`, the only allowed value is `Secondary`.
+        - Having Role explicitly makes the API predictable and consistent with other topologies. In addition, it enables extending localnet to support future role options
+- `PhysicalNetworkName`:
+    -  Required.
+    - Max length 253.
+    - Cannot contain `,` or `:` characters.
+- `MTU`:
+    - Minimum 576 (minimal for IPv4). Maximum: 65536.
+    - When Subnets consist of IPv6 CIDR, minimum MTU should be 1280.
+- `VLAN`:
+  - `Mode`: Can be "Access".
```

Comment:
Can be? You mean it is optional and defaults to something?

Reply:
Just bad wording, I meant to say is should validate the value is "Access".